static_build/scss/components/_generic_menu_bar_selector.scss

```diff
@@ -0,0 +1,165 @@
+.top.menu .selector {
+  height: 100%;
+  width: calc(100% / 3);
+  max-width: 360px;
+
+  @include tvn-theme-normal;
+
+  &:hover {
+    @include tvn-theme-accented;
+    .menu {
+      @extend .menu;
+    }
+  }
+
+  .tab {
+    display: flex;
+    align-items: center;
+    width: 100%;
+    height: 100%;
+    overflow: hidden;
+
+    .section {
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      height: 100%;
+      overflow: hidden;
+    }
+    .info.section {
+      align-items: flex-start;
+      margin-right: 18%;
+    }
+    .icon.section {
+      flex: none;
+      align-items: center;
+      margin: 0 10%;
+    }
+    .icon.section .icon.header > .tvn.icon {
+      width: 35px;
+      height: 35px;
+    }
+    .header {
+      margin: 0;
+      white-space: nowrap;
+      overflow: hidden;
+    }
+    .sub.header {
+      margin-top: 3px;
+    }
+    // this is only for calendar icon
+    .calendar.icon.header > .number {
+      font-size: 1.1em;
+      margin-bottom: 6px;
+    }
+
+    // media queries:
+    @media only screen and (max-width: 767px) {
+      & {
+        justify-content: center;
+      }
+    }
+    @media only screen and (max-width: 480px) {
+      .icon.section .icon.header > .tvn.icon {
+        width: 30px;
+        height: 30px;
+      }
+      // this is only for calendar icon
+      .calendar.icon.header > .number {
+        font-size: 1em;
+        margin-bottom: 6px;
+      }
+    }
+
+  }
+
+  .dropdown.toggle {
+    margin-right: 10%;
+    z-index: 1;
+
+    .tvn.icon {
+      width: $toggle-dropdown-icon-size;
+      height: $toggle-dropdown-icon-size;
+      opacity: $toggle-dropdown-icon-opacity;
+    }
+  }
+
+  .dropdown.menu {
+    display: flex;
+    flex-direction: column;
+    width: 100%;
+    min-width: 280px;
+    min-height: $tvn-fixed-top-menu-height;
+    max-height: 70vh;
+    left: auto;
+    right: 0;
+    border: $tvn-stroke;
+    border-bottom-right-radius: $tvn-content-border-radius;
+    border-bottom-left-radius: $tvn-content-border-radius;
+    box-shadow: $tvn-component-box-shadow;
+    overflow: hidden;
+
+    // media queries:
+    @media only screen and (max-width: 767px) {
+      & {
+        position: fixed;
+        top: $tvn-fixed-top-menu-height !important;
+        min-width: unset;
+        min-height: unset;
+        max-height: calc(100vh - #{$tvn-fixed-top-menu-height});
+      }
+    }
+  }
+
+  .dropdown.menu .content {
+    width: 100%;
+    overflow: auto;
+  }
+
+  .dropdown.menu .footer {
+    flex: none;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 100%;
+    height: $tvn-dropdown-footer-height;
+    border-top: $tvn-stroke;
+    overflow: hidden;
+    @include tvn-theme-normal;
+  }
+
+  .dropdown.menu .list .item {
+    display: flex !important;
```

Comment:
@andela-rsadiq For the most part, I relied on specificity and avoided using`!important` wherever I could. It is rather unfortunate that Semantic UI applies this 'abomination' in some of their style rules which I needed to override with our custom styles.
Given that the only way to override a `!important` is with another `!important`, I was given no other choice but to succumb (reluctantly though) to using this $%#@#$% in ours.

My earnest hope is that my unwilling ethical sacrifice will save others the horrors of doing the same. Long live CSS!!! Long Live Taverna!!!

PS: Please forgive the dramatics, just thought I could spice up the with a little humor ;)